sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/AnsiTypeCoercion.scala

```diff
@@ -288,6 +290,82 @@ object AnsiTypeCoercion extends TypeCoercionBase {
     }
   }
 
+  // Please see the comments in `TypeCoercion.ResolveBinaryArithmetic`.
+  object ResolveBinaryArithmetic extends TypeCoercionRule {
```

Comment:
is it possible to do some code sharing?

Reply:
Yes. Updated the code:
https://github.com/apache/spark/blob/cd21cce9131e17d53b164fdfe8d54a59a265f41a/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala#L1173-L1194